app/models/reporting/decisions.rb

```diff
@@ -5,15 +5,132 @@
 ###
 
 class Reporting::Decisions < ApplicationRecord
+  include ArelHelper
+
   belongs_to :client, foreign_key: :cas_client_id
 
-  scope :started_between, -> (start_date:, end_date:) do
+  scope :started_between, ->(start_date:, end_date:) do
     where(match_started_at: (start_date..end_date))
   end
 
-  scope :ended_between, -> (start_date:, end_date:) do
+  scope :ended_between, ->(start_date:, end_date:) do
     terminated.
-    where(updated_at: (start_date..end_date + 1.day))
+      where(updated_at: (start_date..end_date + 1.day))
+  end
+
+  scope :open_between, ->(start_date:, end_date:) do
+    at = arel_table
+    # Excellent discussion of why this works:
+    # http://stackoverflow.com/questions/325933/determine-whether-two-date-ranges-overlap
+    d_1_start = start_date
+    d_1_end = end_date
+    d_2_start = at[:match_started_at]
+    d_2_end = at[:updated_at]
+    # Currently does not count as an overlap if one starts on the end of the other
+    where(d_2_end.gteq(d_1_start).or(d_2_end.eq(nil)).and(d_2_start.lteq(d_1_end)))
+  end
+
+  scope :on_route, ->(limit) do
+    route_names = Array.wrap(limit).map do |route_id|
+      MatchRoutes::Base.find(route_id).title
+    end
```

Comment:
I _think_ this is the same without the N+1
```suggestion
    route_names =  MatchRoutes::Base.where(Array.wrap(limit)).pluck(:title)
```

Reply:
Unfortunately, `title` is a method overridden in the subclasses of `MatchRoute::Base` so there is no `title` to pluck. But, `find(Array.wrap(limit))` will avoid the N+1.